### PR TITLE
Handle invalid repository url

### LINKED ISSUE #5: `nh -g babel-preset-minify` crashes
```
❯ npm install npm-home
+ npm-home@1.4.1
added 78 packages in 2.241s

❯ ./node_modules/.bin/nh -g babel-preset-minify
Error: Expected a `target`
    at module.exports (/home/lydell/bugs/node_modules/opn/index.js:8:25)
    at packageJson.then.pkg (/home/lydell/bugs/node_modules/npm-home/cli.js:45:12)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

Thanks for this very useful tool, btw!

### PR DESCRIPTION
**Pull request by hafffe**
Fixes #5 
tested with 
```
{
    "name": "chalk",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/chalk/chalk.git"
    },
    "homepage": "https://github.com/chalk/chalk#readme"
};
```

```
{
    "name": "chalk",
    "repository": {
        "type": "git",
        "url": "https://github.com/chalk/chalk"
    },
    "homepage": "https://github.com/chalk/chalk#readme"
};
```

```
{
    "name": "chalk",
    "repository": {
        "type": "git",
        "url": "/chalk/chalk"
    },
    "homepage": "https://github.com/chalk/chalk#readme"
};
```

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -41,7 +42,13 @@ function open(name) {
 	if (cli.flags.github) {
 		return packageJson(name, {fullMetadata: true}).then(pkg => {
 			if (pkg.repository) {
-				const url = githubUrlFromGit(pkg.repository.url);
+				const url = githubUrlFromGit(pkg.repository.url) || pkg.repository.url;
+
+				if (!urlRegex().test(url)) {
+					console.log('Invalid repository URL, opening homepage');
```
Comment (kevva) [inline cli.js]: Maybe use `console.error` here.

Comment (hafffe) [reply cli.js]: Absolutely!

File: cli.js [hunk 2]
```diff
@@ -41,7 +42,13 @@ function open(name) {
 	if (cli.flags.github) {
 		return packageJson(name, {fullMetadata: true}).then(pkg => {
 			if (pkg.repository) {
-				const url = githubUrlFromGit(pkg.repository.url);
+				const url = githubUrlFromGit(pkg.repository.url) || pkg.repository.url;
+
+				if (!urlRegex().test(url)) {
+					console.error('Invalid repository URL, opening homepage');
```
Comment (sindresorhus) [inline cli.js]: The message should be explicit about it being the `repository` property in the package.json being invalid and recommending the user to open an issue/PR on the repo to fix the package.json

Comment (hafffe) [reply cli.js]: Changed :)

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**